File: 프로그래머스/3/43105. 정수 삼각형/정수 삼각형.cpp
// 순서: 아래 -> 위

#include <bits/stdc++.h>
using namespace std;

int solution(vector<vector<int>> triangle) {
    vector<vector<int>> dp(triangle);
    
    for(int i = triangle.size() - 2; i >= 0; i--) {
        for(int j = 0; j < triangle[i].size(); j++) {
            dp[i][j] += max(dp[i + 1][j], dp[i + 1][j + 1]);
        }
    }
    
    return dp[0][0];
}

File: 프로그래머스/3/43105. 정수 삼각형/정수 삼각형.java
import java.util.*;

class Solution {
    public int solution(int[][] triangle) {
        int n = triangle.length;
        int[][] dp = triangle;
        
        for(int i = n - 1; i > 0; i--) {
            for(int j = 0; j < triangle[i].length - 1; j++) {
                dp[i - 1][j] += Math.max(dp[i][j], dp[i][j + 1]);
            }
        }
           
        return dp[0][0];
    }
}